internal/auth0/organization/resource_member_roles_test.go

```diff
@@ -0,0 +1,169 @@
+
+	organization_id = auth0_organization.org.id
+	user_id         = auth0_user.user.id
+
+	roles = [ auth0_role.reader.id ]
+}
+`
+
+const testAccOrganizationMemberRolesCreateWithTwoRoles = `
+resource "auth0_role" "reader" {
+	name = "Test Reader - {{.testName}}"
+}
+
+resource "auth0_role" "writer" {
+	depends_on = [ auth0_role.reader ]
+
+	name = "Test Writer - {{.testName}}"
+}
+
+resource "auth0_user" "user" {
+	depends_on = [ auth0_role.writer ]
+
+	connection_name = "Username-Password-Authentication"
+
+	email    = "{{.testName}}@auth0.com"
+	password = "MyPass123$"
+}
+
+resource "auth0_organization" "org" {
+	depends_on = [ auth0_user.user ]
+
+	name         = "some-org-{{.testName}}"
+	display_name = "{{.testName}}"
+}
+
+resource "auth0_organization_member" "member" {
+	depends_on = [ auth0_organization.org ]
+
+	organization_id = auth0_organization.org.id
+	user_id         = auth0_user.user.id
+
+	lifecycle {
+		ignore_changes = [ roles ]
+	}
+}
+
+resource "auth0_organization_member_roles" "roles" {
+	depends_on = [ auth0_organization_member.member ]
+
+	organization_id = auth0_organization.org.id
+	user_id         = auth0_user.user.id
+
+	roles = [ auth0_role.reader.id, auth0_role.writer.id ]
+}
+`
+
+const testAccOrganizationMemberRolesCreateWithNoRoles = `
+resource "auth0_user" "user" {
+	connection_name = "Username-Password-Authentication"
+
+	email    = "{{.testName}}@auth0.com"
+	password = "MyPass123$"
+}
+
+resource "auth0_organization" "org" {
+	depends_on = [ auth0_user.user ]
+
+	name         = "some-org-{{.testName}}"
+	display_name = "{{.testName}}"
+}
+
+resource "auth0_organization_member" "member" {
+	depends_on = [ auth0_organization.org ]
+
+	organization_id = auth0_organization.org.id
+	user_id         = auth0_user.user.id
+
+	lifecycle {
+		ignore_changes = [ roles ]
+	}
+}
+
+resource "auth0_organization_member_roles" "roles" {
+	depends_on = [ auth0_organization_member.member ]
+
+	organization_id = auth0_organization.org.id
+	user_id         = auth0_user.user.id
+
+	roles = []
+}
+`
+
+func TestAccOrganizationMemberRoles(t *testing.T) {
+	testName := strings.ToLower(t.Name())
+
+	acctest.Test(t, resource.TestCase{
+		Steps: []resource.TestStep{
+			{
+				Config: acctest.ParseTestName(testAccOrganizationMemberRolesCreateWithOneRole, testName),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("auth0_organization_member_roles.roles", "roles.#", "1"),
+				),
+			},
+			{
+				Config: acctest.ParseTestName(testAccOrganizationMemberRolesCreateWithTwoRoles, testName),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("auth0_organization_member_roles.roles", "roles.#", "2"),
+				),
+			},
+			{
+				Config: acctest.ParseTestName(testAccOrganizationMemberRolesCreateWithOneRole, testName),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("auth0_organization_member_roles.roles", "roles.#", "1"),
+				),
+			},
```

Comment:
Isn't this the same as the first test?

Reply:
It might seem that way, but it's actually not, as the order here matters. In the first test we will perform a `CREATE`, and in this one an `UPDATE`.

Reply:
Update: nvm had to update it anyway as the roles need to always exist before removing them from an org member, otherwise we produce inconsistent results. Check [`1880a2c` (#617)](https://github.com/auth0/terraform-provider-auth0/pull/617/commits/1880a2c6ed596eef9b62e7c0742f1d87961e3fb4)